src/infer_request.cc

```diff
@@ -682,6 +714,72 @@ InferenceRequest::Normalize()
 {
   const inference::ModelConfig& model_config = model_raw_->Config();
 
+  // Fill metadata for raw input
+  if (!raw_input_name_.empty()) {
+    if ((original_inputs_.size() != 1) || (model_config.input_size() != 1)) {
+      return Status(
+          Status::Code::INVALID_ARG,
+          "Raw request must only have 1 input to be deduced but got " +
+              std::to_string(model_config.input_size()) +
```

Comment:
Maybe should add the original_inputs_.size() to error message? Since the if statement includes both conditions, it's possible the model config has 1 input but the original inputs do not.